packages/react/src/components/DataTable/DataTable.tsx

```diff
@@ -0,0 +1,112 @@
+import { DataTableRow } from '../DataTableRow';
+import { Pagination } from '../Pagination';
+import { DataTableContext } from './DataTable.context';
+import { StyledDataTable } from './DataTable.styles';
+import type { DataTableElement, DataTableProps } from './DataTable.types';
+import { StyledDataTableOverflowContainer } from '../DataTableOverflowContainer/DataTableOverflowContainer.styles';
+import { DataTableContent } from '../DataTableContent';
+
+export const DataTable = React.forwardRef((props, forwardedRef) => {
+  const {
+    as,
+    children,
+    className: classNameProp,
+    columns,
+    css,
+    currentPage,
+    isDense = false,
+    onPageChange,
+    onSort,
+    rows,
+    totalRowCount,
+    ...other
+  } = props;
+
+  const className = cx('manifest-data-table', classNameProp);
+
+  const [sortDirection, setSortDirection] = React.useState<'asc' | 'desc'>('asc');
+
+  const [sortedColumn, setSortedColumn] = React.useState<string>();
+
+  const context = React.useMemo(() => ({ isDense }), [isDense]);
+
+  const handleSort = (column: string) => () => {
+    const isAsc = sortedColumn === column && sortDirection === 'asc';
+
+    setSortDirection(isAsc ? 'desc' : 'asc');
+    setSortedColumn(column);
+    onSort(column, isAsc);
+  };
+
+  React.useEffect(() => {
+    if (process.env.NODE_ENV === 'development') {
+      const numberOfLeftColumns = columns.filter((column) => column.isPinnedLeft).length;
+      if (numberOfLeftColumns > 1) {
+        // eslint-disable-next-line no-console
+        console.warn('Only one column can be pinned left');
+      }
+
+      const numberOfRightColumns = columns.filter((column) => column.isPinnedRight).length;
+      if (numberOfRightColumns > 1) {
+        // eslint-disable-next-line no-console
+        console.warn('Only one column can be pinned right');
+      }
+    }
+  }, [columns]);
+
+  return (
+    <StyledDataTable {...other} ref={forwardedRef} as={as} className={className} css={css}>
+      <DataTableContext.Provider value={context}>
+        <StyledDataTableOverflowContainer>
+          <DataTableContent>
+            <DataTableHeader>
+              <DataTableRow>
+                {columns.map((column) => (
+                  <DataTableColumn
+                    key={column.key}
+                    isActive={sortedColumn === column.key}
+                    isPinnedLeft={column.isPinnedLeft}
+                    isPinnedRight={column.isPinnedRight}
+                    isSortable={column.isSortable}
+                    sortDirection={sortDirection}
+                    onClick={column.isSortable ? handleSort(column.key) : undefined}
+                  >
+                    {column.header}
+                  </DataTableColumn>
+                ))}
+              </DataTableRow>
+            </DataTableHeader>
+            <DataTableBody>
+              {rows.map((row, idx) => (
+                <DataTableRow key={`row-${String(idx)}`}>
```

Comment:
open to suggestions on how not to do this. uuid or req some row schema that guarantees a key/id